lib/arith.gd

```diff
@@ -966,8 +966,14 @@ DeclareCategoryCollections("IsZDFRECollection");
 ##  <Filt Name="IsMatrix" Arg='obj' Type='Category'/>
 ##
 ##  <Description>
-##  A <E>matrix</E> is a list of lists of equal length whose entries lie in a
-##  common ring.
+##  By convention <E>matrix</E> is a list of lists of equal length whose
+##  entries lie in a common ring.
+##  <P/>
+##  For technical reasons laid out at the top of Chapter <Ref Chap="Matrices"/>,
+##  the filter <Ref Filt="IsMatrix"/> is a synonym for a table of ring elments,
+##  (see <Ref Filt="IsTable"/> and <Ref Filt="IsRingElement"/>)
+##  If necessary, <Ref Prop="IsRectangularTable"/> can be used to test whether
+##  an object is a matrix manually.
```

Comment:
I agree with @fingolfin that this could be clearer. Might it not be clearest to make the distinction between what is mathematically a matrix and what satisfies `IsMatrix`? 

If I read, "This means that sometimes IsMatrix returns true for things were it really shouldn't...", then I would think that this meant that `IsMatrix` is broken, and does not return the intended result sometimes.